src/App.tsx

```diff
@@ -15,36 +23,82 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function sortGoods(goods: string[], sortField: SortType, reversed: boolean) {
+  const sortedGoods = [...goods].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return reversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState(false);
+
+  const goods = sortGoods(goodsFromServer, sortField, reversed);
+
+  function reset() {
+    setSortField(SortType.None);
+    setReversed(false);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.Alphabet,
+          })}
+          onClick={() => setSortField(SortType.Alphabet)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.Length,
+          })}
+          onClick={() => setSortField(SortType.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', 'is-warning', {
+            'is-light': !reversed,
+          })}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== SortType.None || reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
```

Comment:
The conditional rendering of the 'Reset' button is correct. However, ensure that the logic for showing this button is consistent with the intended functionality. It should appear only when there is a sort type selected or the list is reversed.